apps/admin/backend/src/tabulation/full_results.test.ts

```diff
@@ -358,65 +361,117 @@ test('tabulateElectionResults - write-in handling', async () => {
-  });
-  store.adjudicateWriteIn({
-    writeInId: writeIn2!.id,
-    type: 'invalid',
-  });
-  store.adjudicateWriteIn({
-    writeInId: writeIn3!.id,
-    type: 'official-candidate',
-    candidateId: 'Obadiah-Carrigan-5c95145a',
-  });
-  store.adjudicateWriteIn({
-    writeInId: writeIn4!.id,
-    type: 'official-candidate',
-    candidateId: 'Abigail-Bartlett-4e46c9d4',
-  });
+  adjudicateWriteIn(
+    {
+      writeInId: writeIn1!.id,
+      type: 'invalid',
+    },
+    store
+  );
+  adjudicateWriteIn(
+    {
+      writeInId: writeIn2!.id,
+      type: 'invalid',
+    },
+    store
+  );
+  adjudicateWriteIn(
+    {
+      writeInId: writeIn3!.id,
+      type: 'official-candidate',
+      candidateId: 'Obadiah-Carrigan-5c95145a',
+    },
+    store
+  );
+  adjudicateWriteIn(
+    {
+      writeInId: writeIn4!.id,
+      type: 'official-candidate',
+      candidateId: 'Abigail-Bartlett-4e46c9d4',
+    },
+    store
+  );
   const writeInCandidate1 = store.addWriteInCandidate({
     electionId,
     contestId: candidateContestId,
     name: 'Mr. Pickles',
   });
-  store.adjudicateWriteIn({
-    writeInId: writeIn5!.id,
-    type: 'write-in-candidate',
-    candidateId: writeInCandidate1.id,
-  });
+  adjudicateWriteIn(
+    {
+      writeInId: writeIn5!.id,
+      type: 'write-in-candidate',
+      candidateId: writeInCandidate1.id,
+    },
+    store
+  );
   const writeInCandidate2 = store.addWriteInCandidate({
     electionId,
     contestId: candidateContestId,
     name: 'Ms. Tomato',
   });
-  store.adjudicateWriteIn({
-    writeInId: writeIn6!.id,
-    type: 'write-in-candidate',
-    candidateId: writeInCandidate2.id,
-  });
+  adjudicateWriteIn(
+    {
+      writeInId: writeIn6!.id,
+      type: 'write-in-candidate',
+      candidateId: writeInCandidate2.id,
+    },
+    store
+  );
 
-  // if we don't specify we need the WIA data, results are the same
+  // if we don't specify we need the detailed WIA data, undervotes still reflect the invalid write-ins
+  const overallResultsScreenWiaNoDetail = (
+    await tabulateElectionResults({
+      electionId,
+      store,
+    })
+  )[GROUP_KEY_ROOT];
+  assert(overallResultsScreenWiaNoDetail);
+  const partialExpectedResultsScreenWiaNoDetail = buildElectionResultsFixture({
+    election,
+    cardCounts: {
+      bmd: 0,
+      hmpb: [184],
+    },
+    contestResultsSummaries: {
+      [candidateContestId]: {
+        type: 'candidate',
+        ballots: 184,
+        overvotes: 30,
+        undervotes: 14,
+        officialOptionTallies: {
+          'Abigail-Bartlett-4e46c9d4': 56,
+          'Elijah-Miller-a52e6988': 56,
+          'Isaac-Hill-d6c9deeb': 56,
+          'Jacob-Freese-b5146505': 56,
+          'Mary-Baker-Eddy-350785d5': 58,
+          'Obadiah-Carrigan-5c95145a': 60,
+          'Samuel-Bell-17973275': 56,
+          'Samuel-Livermore-f927fef1': 56,
+          'write-in': 54,
+        },
+      },
+    },
+    includeGenericWriteIn: true,
+  });
```

Comment:
Previously we had the problem that got fixed for free. You can either tabulate without bringing in write-in adjudication data or not. if you tabulated cast vote records with write-in data, it would update with the list of actual write-ins and update the undervote count. And if you tabulated cast vote records with write-in data, it would not fix the undervote count. But a more reasonable use case than the latter is that you don't need the fine write-in data, but you at least want the invalid ones reflected as undervotes. Because the undervote accounting has moved a layer up, into the cast vote record tabulation, the undervotes will always be correct. A number of these test updates reflect that change.